realm/realm-library/src/main/java/io/realm/MutableRealmInteger.java

```diff
@@ -57,104 +68,92 @@ public Long get() {
-
-        /**
-         * Inject the proxy state into this managed MutableRealmInteger.
-         *
-         * @param proxyState Proxy state object.  Contains refs to Realm and Row.
-         * @param columnIndex The index of the column that contains the MutableRealmInteger.
-         * @return a managed MutableRealmInteger.
-         */
-        Managed(ProxyState<? extends RealmObject> proxyState, long columnIndex) {
-            this.proxyState = proxyState;
-            this.realm = proxyState.getRealm$realm();
-            this.row = proxyState.getRow$realm();
-            this.columnIndex = columnIndex;
-        }
+    abstract static class Managed<T extends RealmModel> extends MutableRealmInteger {
+        protected abstract ProxyState<T> getProxyState();
+
+        protected abstract long getColumnIndex();
 
         @Override
-        public boolean isManaged() {
+        public final boolean isManaged() {
             return true;
         }
 
         @Override
-        public boolean isValid() {
-            return !realm.isClosed() && row.isAttached();
+        public final boolean isValid() {
+            return !getRealm().isClosed() && getRow().isAttached();
         }
 
         @Override
-        public Long get() {
-            return value;
+        public final Long get() {
+            Row row = getRow();
+            row.checkIfAttached();
+            long columnIndex = getColumnIndex();
+            return (row.isNull(columnIndex)) ? null : row .getLong(columnIndex);
         }
 
-
-//        // Template code:
-//        if (proxyState.isUnderConstruction()) {
-//            if (!proxyState.getAcceptDefaultValue$realm()) {  // Wat?
-//                return;
-//            }
-//
-//            row.getTable().setLong(columnIndex, row.getIndex(), value, true);
-//            return;
-//        }
-//        row.setLong(columnIndex,value);
         @Override
-        public void set(Long value) {
-            realm.checkIfValidAndInTransaction();
-            this.value = value;
+        public final void set(Long value) {
+            ProxyState proxyState = getProxyState();
+            proxyState.getRealm$realm().checkIfValidAndInTransaction();
+
+            if (!proxyState.isUnderConstruction()) {
+                setValue(value, false);
+                return;
+            }
+
+            if (!proxyState.getAcceptDefaultValue$realm()) {
+                return;
+            }
+
+            setValue(value, true);
         }
 
         @Override
-        public void increment(long inc) {
-            Long val = get();
-            if (val == null) {
-                throw new NullPointerException("Attempt to increment a null valued MutableRealmInteger");
-            }
-            set(val + inc);
+        public final void increment(long inc) {
+            getRealm().checkIfValidAndInTransaction();
+            Row row = getRow();
+            row.getTable().incrementLong(getColumnIndex(), row.getIndex(), inc);
```

Comment:
Yes.  I answered this above, for @beeender .  There is a task, [here](https://github.com/realm/realm-java/issues/4266#issuecomment-313474899) to make sure this all works for the dynamic API.